src/background/bgPopupHandler.ts

```diff
@@ -4,17 +4,24 @@ import type {
   IPopupProps,
   PopupType,
 } from '@/types';
-import {
-  POPUP_TYPE_CONNECT,
-  IS_EXTENSION,
-} from '@/constants';
 
-interface IPopupConfig {
-  actions: Pick<IPopupProps, 'resolve' | 'reject'>;
+interface IPopupConfigNoActions {
+  id: string;
   props: Omit<IPopupProps, 'resolve' | 'reject'>;
 }
 
-const popups: Dictionary<IPopupConfig> = {};
+/**
+ * We need to duplicate these constants here because
+ * importing from constants.ts causes error in background script
+ * due to undefined window object
+ */
+const PLATFORM = process.env.PLATFORM as 'web' | 'extension' | 'ionic';
+const RUNNING_IN_TESTS = !!process.env.RUNNING_IN_TESTS;
+const IS_EXTENSION = PLATFORM === 'extension' && !RUNNING_IN_TESTS;
```

Comment:
If the background lacks of `window` object I suggest to use `?.` to pick any window property. Eg.: `window.name.startsWith('popup-')` => `window?.name?.startsWith('popup-')`

Reply:
Unfortunately this is not possible. `window` is can't even be present at the background script.
If it is we get the following error, even with something like `window?.name?.startsWith('popup-')`

![image](https://github.com/superhero-com/superhero-wallet/assets/49337371/88cbe392-800e-4302-af83-db9592b4dad7)